diff -ru src/lcov/lcov-1.10/bin/install.sh src/lcov/lcov-1.10-portz-patch/bin/install.sh
--- src/lcov/lcov-1.10/bin/install.sh	2012-12-27 12:21:45.248849756 +0100
+++ src/lcov/lcov-1.10-portz-patch/bin/install.sh	2012-12-27 12:22:49.582848491 +0100
@@ -33,8 +33,8 @@
   local SOURCE=$1
   local TARGET=$2
   local PARAMS=$3
-
-  install -p -D $PARAMS $SOURCE $TARGET
+  mkdir -p `dirname $TARGET`
+  install -p $PARAMS $SOURCE $TARGET
 }
 
 
