--- pcre-8.30/Makefile.am	2012-05-27 07:10:06.048083230 +0200
+++ pcre-8.30-build-fix/Makefile.am	2012-05-27 07:05:14.610208929 +0200
@@ -339,7 +339,7 @@
 pcretest_LDADD = $(LIBREADLINE)
 if WITH_PCRE8
 pcretest_SOURCES += pcre_printint.c
-pcretest_LDADD += libpcreposix.la
+pcretest_LDADD += libpcreposix.la libpcre.la
 endif # WITH_PCRE8
 if WITH_PCRE16
 pcretest_SOURCES += pcre16_printint.c
@@ -352,7 +352,7 @@
 bin_PROGRAMS += pcregrep
 pcregrep_SOURCES = pcregrep.c
 pcregrep_LDADD = $(LIBZ) $(LIBBZ2)
-pcregrep_LDADD += libpcreposix.la
+pcregrep_LDADD += libpcreposix.la libpcre.la
 endif # WITH_PCRE8
 
 EXTRA_DIST += \
